zcash_client_sqlite/src/lib.rs

```diff
@@ -1285,6 +1301,32 @@ mod tests {
             .unwrap();
     }
 
+    #[cfg(feature = "unstable")]
+    pub(crate) fn store_in_fsblockdb<P: AsRef<Path>>(
+        fsblockdb_root: P,
+        cb: &CompactBlock,
+    ) -> BlockMeta {
+        use std::io::Write;
+
+        let meta = BlockMeta {
+            height: cb.height(),
+            block_hash: cb.hash(),
+            block_time: cb.time,
+            sapling_outputs_count: cb.vtx.iter().map(|tx| tx.outputs.len() as u32).sum(),
+            orchard_actions_count: cb.vtx.iter().map(|tx| tx.actions.len() as u32).sum(),
+        };
+
+        let blocks_dir = fsblockdb_root.as_ref().join("blocks");
+        let block_path = meta.block_file_path(&blocks_dir);
+
+        File::create(block_path)
+            .unwrap()
+            .write_all(&cb.encode_to_vec())
+            .unwrap();
+
+        meta
+    }
+
```

Comment:
Should we actually make a method like this part of the `BlockStore` interface?

Reply:
Maybe in future when we need it from Rust; for now this is logic that the mobile SDKs will be implementing themselves.